### DISCUSSION TITLE
How can i work with localhost APIs?

### DISCUSSION THREAD

**Discussion by juaiglesias** [hoppscotch/hoppscotch#2051] [Q&A] [answered]
Hello everyone! 
First of all, thanks for this space. I wanted to ask how it's possible to work with localhost APIs, I couldn't find enought doc about this and only configuring the proxy in the app (https://hoppscotch.io/es/settings) doesn't worked for me.

**Comment by liyasthomas** [comment, marked as answer]
You should install our [Browser Extension](https://github.com/hoppscotch/hoppscotch-extension) and switch the interceptor to Browser Extension from [Settings](https://hoppscotch.io/settings) or bottom bar.

---

Step 1: Install our [Browser Extension](https://github.com/hoppscotch/hoppscotch-extension)

---

Step 2: Click on the Hoppscotch extension icon from Browser's toolbar to add your localhost origins to the Active Origins list.
ex: `http://localhost:3000` or `http://localhost:8080`

<img width="1440" alt="Screenshot 2021-12-30 at 7 46 27 PM" src="https://user-images.githubusercontent.com/10395817/147759783-cfdc3079-80eb-49fa-807c-c873521eef50.png">

---

Step 3: Refresh the app

---

Step 4: Change interceptor to Browser Extension

App's bottom bar - Interceptor menu:

<img width="1440" alt="Screenshot 2021-12-30 at 7 36 27 PM" src="https://user-images.githubusercontent.com/10395817/147758989-46606de2-b88a-4c18-a859-1fa2ebaa5ea7.png">

or Settings page - Interceptor > Extensions section:

<img width="1440" alt="Screenshot 2021-12-30 at 7 39 47 PM" src="https://user-images.githubusercontent.com/10395817/147759226-1a3cf0b9-a83c-4a68-b4a8-173977858c9b.png">

NOTE: Refresh the app after installing the extension. You should be able to see the extension version under the Browser Interceptor menu.

**Reply by juaiglesias** [reply]
Hello Thomas, thank you for the fast answer.
I've done all that and it's still not working, checking the network activity when a do a request, I get this:
![image](https://user-images.githubusercontent.com/37712048/147760956-40648ce5-2844-4e1c-90f3-56a32ca45d89.png)
Not much network activity and a console error, might be that?

**Reply by liyasthomas** [reply]
Can you share how to reproduce this error?

You can copy the request (Save menu > Copy link) link and paste it here as a reply.

Screenshot:

<img width="1440" alt="Screenshot 2021-12-30 at 8 17 09 PM" src="https://user-images.githubusercontent.com/10395817/147761955-d0574429-79de-4b1f-8727-56bd32292d7b.png">

**Reply by juaiglesias** [reply]
I found out the problem thanks to your answer. Here is the link: https://hopp.sh/r/cqw1Kb2CciW2
I think that may be something related to the fact that I imported the requests from a postman collection. If i generate them fresh (using the Refresh option) I have not problems at all.

**Reply by liyasthomas** [reply]
Noted. We're working on a fully revamped set of migration tools (including Postman, Insomnia, OpenAPI, etc). Stay tuned for the product updates. We'll look into improving the stability of the Postman Collection parser.

**Reply by juaiglesias** [reply]
Cool! Thanks again for helping me woked this out.

**Reply by Olyno** [reply]
Hi @liyasthomas 👋🏻 
I'm a bit late, but a question was bothering me: why do I need an additional extension to access the localhost? What is the technical constraint to use this extension? Why not just add a parameter in the options?

I am quite ignorant in the method used, thus my question 😅

**Reply by npearson72** [reply]
@Olyno and @liyasthomas I'm curious about this too. Is there no way to white list an origin from within the app itself?

**Reply by daryllreillo** [reply]
@liyasthomas,
thank you so much for this. I've been searching for a solution for weeks now and fortunately found your solution.
You're a lifesaver.

**Reply by tauqeer1040** [reply]
I'm working with a Linux x32 machine and this seems to be one of the few ways to test localhost. Postman doesn't have an x32 desktop agent.  This is a fantastic help for my learning journey. Thanks a lot a lot.

**Reply by EAbimelec** [reply]
Thanks for the explanation it works in brave.

**Reply by Piyushgupta-1** [reply]
Thanks..

**Reply by IOrlandoni** [reply]
thanks to i will try it at [localhost -](https://localhost.co/).

**Reply by sachin-bi** [reply]
**thanks..** it works fine

**Reply by Aayanshi** [reply]
Thankyou So much  it was really helpful

**Reply by KingShivamX** [reply]
thanks dude

**Reply by somnathshil** [reply]
I have tried to run my localhost in incognito but it's failed to run . Anybody can help me out from this problem?
![IMG20240720201154](https://github.com/user-attachments/assets/72c1cca9-a203-4d78-8db7-a5f7520159c8)
![IMG20240720204731](https://github.com/user-attachments/assets/b0a553fe-14f2-4858-9788-635c2b309683)

**Reply by mansi-halkude** [reply]
reload the page

**Comment by Bowrna** [comment]
I tried to use hoppscotch to test the API in localhost using the above method you have suggested. But it didn't work for me. Is there anything that I missed out on? I refreshed the app as you suggested @liyasthomas 

<img width="1440" alt="Screenshot 2022-09-18 at 3 46 27 PM" src="https://user-images.githubusercontent.com/10162465/190897198-48bfb144-360a-4f76-9afc-a72de340931f.png">

**Reply by liyasthomas** [reply]
Can you try adding the local IP address alone to the Active origins list - not with the port number?

**Reply by Bowrna** [reply]
Still, the same issue occurs for me @liyasthomas 
I remember it was working for me previously and I have the history of it working. I am using Mac M1. 
<img width="1440" alt="Screenshot 2022-09-18 at 9 27 47 PM" src="https://user-images.githubusercontent.com/10162465/190916388-494a5755-23c3-49f9-84e8-e9c434e9f1fc.png">

**Reply by rafmst** [reply]
Same with me

**Reply by yuaoyoung** [reply]
i just had same problem

**Reply by AndrewBastin** [reply]
@Bowrna sorry you are experiencing this, I would suggest to try restarting your browser, and seeing if the issue persists. If it does, please can you open an issue on github.com/hoppscotch/hoppscotch-extension so we can track this issue.

**Reply by Karthik-Sunil-K** [reply]
It Happens is because of somany tabs are opened in your browser follow all the above instructions and just restart your browser the issue will fix :)

**Reply by pacific-prashant** [reply]
This extension is not working for me. 
<img width="1080" alt="image" src="https://github.com/hoppscotch/hoppscotch/assets/83504276/7e165b68-aef1-450e-b261-73ea6071df50">
anyone who can help

**Reply by mansiprajapati01** [reply]
instead of chrome extension you can use edge extension it's works for me.

**Reply by pacific-prashant** [reply]
okk...are you doing delta? anyone here?

**Reply by AnkanRoy321** [reply]
yes delta 3.0

**Reply by Himanshu-Yadav897** [reply]
Yes , I am bro

**Reply by Nancydhattarwal** [reply]
but to use edge extension , here their is no option for that

**Reply by Mderra123** [reply]
**Found the solution!**

Navigate to your localhost and right click the Hoppscotch Icon.

Now you should see this:

![image](https://github.com/hoppscotch/hoppscotch/assets/60524332/43c79faa-47cd-448f-b427-3657207fe0eb)

Change it from "Only when clicked" to "Always allow on "localhost:3000"

Same with Hoppscotch site just in case:

![image](https://github.com/hoppscotch/hoppscotch/assets/60524332/7e7c48ea-f9d6-42c1-9225-c45338450f04)

**Reply by KaranSJ22** [reply]
Where do I find this

**Reply by gorvsngh** [reply]
I'm not getting it please help

**Reply by vishal-gawade-031** [reply]
thank's bhai

**Comment by codewithjk** [comment]
use private window to avoid some sort of errors

**Reply by KavishParmar** [reply]
extension is not working on localhost in my pc , can anyone has another way to make it work ,or resolve this issue

**Comment by CreateChance** [comment]
As @liyasthomas said, it works with chrome extensions, don't forget turn on [use browser extension to send requests] option on settings pannel. :)

**Comment by needhourger** [comment]
I tried the same operation on firefox, using browser extensions, using private window, close all other tabs, and restart my firefox browser. None of these works for my firefox to test on localhost:5000.

However, chrome works.

**Reply by zavbala** [reply]
Yup, chrome is working 🙌

**Reply by chizaki-ki** [reply]
Apparently you need to enable "Access your data for all websites" permission for this extension for firefox side .
It finally works on firefox after I enable this permission.
<img width="693" alt="image" src="https://github.com/hoppscotch/hoppscotch/assets/67811808/fad2d08b-7d15-4905-893d-af4bfb99b21e">

(Also add origin since I'm running self-host version)

**Reply by rschristian** [reply]
Thanks for adding this! Wouldn't have thought that was needed (and seems a little sketchy, but whatever).

**Comment by jase52476** [comment]
How do you use localhost with the desktop version?
There's no Active Origins setting available that I can find.

**Comment by ThisIs-Developer** [comment]
### Install the browser extension from the below links:

> [Chrome](https://chrome.google.com/webstore/detail/hoppscotch-browser-extens/amknoiejhlmhancpahfcfcfhllgkpbld?hl=en) - **Chrome Web Store**
> [Firefox](https://addons.mozilla.org/en-US/firefox/addon/hoppscotch) - **Add-ons for Firefox**

**Comment by raymondgp** [comment]
Is there any way to get this working without the browser extension? Our Cyber team is not really onboard with adding an extension that can bypass CORS. We have implemented the internal hoppscotch proxy as interceptor but we still see this problem.

**Reply by liyasthomas** [reply]
Have you tried the desktop app - https://hoppscotch.com/download

**Comment by sanketdev1234** [comment]
not working

**Comment by SADHANA-MAURYA** [comment]
my localhost not work in the hoppschotch

**Comment by sanketdev1234** [comment]
Use http://localhost:8080    ,   not https

**Comment by SADHANA-MAURYA** [comment]
My hoppscotch extention not work

**Reply by KavishParmar** [reply]
have a found a solution of it , i also have the same issue

**Comment by SADHANA-MAURYA** [comment]
![Screenshot 2024-12-20 092259](https://github.com/user-attachments/assets/8ec172c6-47a9-48c8-8aff-dbb0d49aed32)
this problm

**Comment by SADHANA-MAURYA** [comment]
not work localhost in hoppschotch

**Comment by liyasthomas** [comment]
You should use the HTTP protocol instead of the HTTPS protocol while
testing localhost.

- Liyas

**Reply by SADHANA-MAURYA** [reply]
yes

**Reply by SADHANA-MAURYA** [reply]
not work i can not understand

**Reply by SADHANA-MAURYA** [reply]
exactly work this app

**Reply by SADHANA-MAURYA** [reply]
please suggest me any viedo

**Reply by liyasthomas** [reply]
@SADHANA-MAURYA if you're still having trouble, download our desktop app with better compatibility with locally served APIs.

https://hoppscotch.com/download

**Comment by sanketdev1234** [comment]
Enable interceptor to browser extension

**Comment by sanketdev1234** [comment]
And follow the process that is mention on github

**Reply by SADHANA-MAURYA** [reply]
not add to extention in hoppscotch

**Comment by SADHANA-MAURYA** [comment]
Not working 😭

**Comment by Nishantsingh280911** [comment]
Just use postman with postman agent tried alot with hoppscotch but didn't work

**Comment by sanketdev1234** [comment]
Yes do , http instead of https

**Comment by yashVeer216** [comment]
use http instead of https

**Reply by igxhemant** [reply]
correct solution
